lazyflow/operators/tiktorch/classifier.py

```diff
@@ -91,104 +91,148 @@ def name(self) -> str:
         return self.__session.name
 
     @property
-    def input_axes(self) -> List[str]:
+    def input_names(self) -> Sequence[str]:
+        """Get names/ids for all model inputs"""
+        return self.__session.inputNames
+
+    @property
+    def output_names(self) -> Sequence[str]:
+        """Get names/ids for all model outputs"""
+        return self.__session.outputNames
+
+    @property
+    def input_axes(self) -> Sequence[str]:
+        """Get axes for all model inputs
+
+        linked via index to `input_names`
+        """
         return self.__session.inputAxes
 
     @property
-    def output_axes(self) -> List[str]:
+    def output_axes(self) -> Sequence[str]:
+        """Get axes for all model inputs
+
+        linked via index to `output_names`
+        """
         return self.__session.outputAxes
 
-    def get_output_shapes(self) -> List[List[Dict[str, int]]]:
+    def get_output_shapes(self) -> Dict[str, Sequence[Dict[str, int]]]:
```

Comment:
Is it possible and/or makes sense to simplify this method and `get_input_shapes` by extracting the inner loop code into something like `input_shapes_of(name: str, axes: str) -> Generator[Dict[str, int]]`? It seems like we don't do much across different input/output names.

Reply:
Good point. I'll leave it like this for now, but will keep this in mind once dealing with a multi-input/multi-output model. It would for sure be more efficient if one accesses only a single input/output.

Reply:
I was actually thinking more about readability, not performance: this shouldn't be a performance bottleneck, but tracking multiple shapes/axes and their potential relationships was a little bit difficult for me to do.